oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/LoginEndpoint.java

```diff
@@ -50,4 +57,12 @@ public void setPassword(String password) {
     this.password = password;
   }
 
+  public String getErrormessage() {
+	return errormessage;
+  }
+
+  public void setErrormessage(String errormessage) {
```

Comment:
You are not posting the message from the login page. The username and password setters are used by sitebricks to set them but the error message is not posted so it is never set or get.

Reply:
The LoginPage redirects to itself with &errormessage=Invalid+username+or+password and that sets the field.